src/main/scala/is/hail/asm4s/CodeWithRefs.scala

```diff
@@ -0,0 +1,26 @@
+package is.hail.asm4s
+
+import scala.language.implicitConversions
+
+case class CodeWithRefs[T](action: FunctionBuilder[_] => T) {
```

Comment:
Rename.  CodeWithRefs sounds like it should extend/contain a Code.  It's really FunctionBuilderReader.  That's not great, either.  It's the thing that actually generates the byte code.  CodeGen?